libs/ACH.js

```diff
@@ -42,16 +43,22 @@ class ACH {
     this.token = token || getTokenPath(url, doctypes)
   }
 
-  connect() {
+  async connect() {
     log.debug('Connecting to ' + this.url)
-    return getClient(this.token, this.url, this.doctypes)
-      .then(client => {
-        this.client = client
-      })
-      .catch(err => {
-        log.warn('Could not connect to' + this.url)
-        throw err
+    try {
+      /* NOTE: in the future, we expect to only use cozy-client */
+      this.client = await getClient(this.token, this.url, this.doctypes)
+      this.cozyClient = await createClientInteractive({
+        uri: this.url,
+        scope: this.doctypes,
+        oauth: {
+          softwareID: 'ACH'
+        }
```

Comment:
Just to be sure, I see that connect was returning the promise returned by getClient; is there any place where we use the result of connect() ? If yes, we should return `this.client` I guess.

Something like `const client = await ach.connect()`

Reply:
No, all the occurrences I found considered that the returned promise is empty, like [here](https://github.com/cozy/ACH/blob/master/cli.js#L194-L195), [here](https://github.com/cozy/ACH/blob/master/repl.js#L41) [here](https://github.com/cozy/ACH/blob/master/libs/importData.js#L200) or [here](https://github.com/cozy/ACH/blob/master/libs/runBatch.js#L23) and `ach.client` is directly used afterwards

Reply:
Thanks ofr the check